mongodbatlas/resource_mongodbatlas_database_user.go

```diff
@@ -49,14 +49,20 @@ func resourceMongoDBAtlasDatabaseUser() *schema.Resource {
 				Type:          schema.TypeString,
 				Optional:      true,
 				Sensitive:     true,
-				ConflictsWith: []string{"x509_type"},
+				ConflictsWith: []string{"x509_type", "ldap_auth_type", "aws_iam_type"},
 			},
 			"x509_type": {
 				Type:         schema.TypeString,
 				Optional:     true,
 				Default:      "NONE",
 				ValidateFunc: validation.StringInSlice([]string{"NONE", "MANAGED", "CUSTOMER"}, false),
 			},
+			"ldap_auth_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "NONE",
+				ValidateFunc: validation.StringInSlice([]string{"NONE", "USER", "GROUP"}, false),
```

Comment:
looks good to me, just one test, create user let's say x509_type. and perform a refresh of the provider to the one that has this feature. 

I would expect to only see an in-place update, and no other complain from terraform